go/parquet/pqarrow/helpers.go

```diff
@@ -38,6 +38,8 @@ func releaseArrayData(data []arrow.ArrayData) {
 
 func releaseColumns(columns []arrow.Column) {
 	for _, col := range columns {
-		col.Release()
+		if col.Data() != nil { // data can be nil due to the way columns are constructed in ReadRowGroups
```

Comment:
Can you please move this `nil` check to within the implementation of `Column.Release()` (and `Column.Retain()` as well) to help prevent this scenario from coming up elsewhere?

Reply:
I'm not sure about this change; with it, I feel that it would imply that a `Column` with `nil` `data` is a valid instance.
The issue with `ReadRowGroups` is that it prepares a `[]Column`, which in error cases can be partially initialized, and then needs to release it.
So, conditional release is really a concern of the `[]arrow.Column` type, not of the `Column` type.

But I don't have any experience with this code base, so I'm also happy to make the change you suggest.

Reply:
We do it in a few other places, but your reasoning here makes sense to me. I'm happy leaving it as it is.